### DISCUSSION TITLE
Builds deprecated, how to enable custom builder?

### DISCUSSION THREAD

**Discussion by thgh** [vercel/vercel#4542] [General] [answered]
Hi everyone,

The built-in Sapper builder is static only while the Next.js builder can do SSR.
Does this new restriction mean we can no longer deploy actual Sapper projects?
Can the Vercel team perhaps take over `vercel-sapper` and use it as the default Sapper builder?
Is there a place where these major breaking changes are communicated beforehand?

https://vercel.com/docs/configuration?query=builds#project/builds
https://github.com/thgh/vercel-sapper/issues/35

**Comment by thgh** [comment]
Looks like not everyone gets this error: `Error! The property `builds` is not allowed in vercel.json – please remove it.`

**Comment by mcsdevv** [comment, marked as answer]
Hello @thgh, whilst the `builds` property is deprecated it can still be used and will continue to be supported. We recommend static-first wherever possible but you could implement SSR with Sapper using a custom Builder.

**Reply by thgh** [reply]
OK, turns out the real issue is that you can't use `builds` and `functions` together. Should I create an issue for that?

**Reply by mcsdevv** [reply]
Hello, this is by design, it is not intended that both be used together.
